### ISSUE TITLE
Add ID to the NFT UTXO model

### ISSUE THREAD

**Issue by samuel-uniris** [archethic-foundation/archethic-node#303]
### Problem to solve

In Bitcoin UTXO model, to identify which UTXO from which transaction we are using, a transaction ID and UTXO ID is used.

Also, in order to support multiple NFT UTXO and specific consumption of them, we need a way to identify the one to use.

This is particularly necessary in NFT collection minting, where we are using UTXO for each token and want to send them individually to other wallets.

### Solution

The proposal would be to add an UTXO's ID and NFT's ID in the transaction data structure to resolve this point.
So a transaction could be built to use a specific NFT UTXO to consume.

This would involve:
- [x] Update transaction data structure for NFT Ledger and UTXO
- [x] Update the ledger operations consumption algorithm

**Comment by AmodD** [comment]
Hey team! Please [add your planning poker estimate](https://app.zenhub.com/workspaces/Product-and-Design-Workspace-61efe71b9083a90018250d89/issues/archethic-foundation/archethic-node/303?planning-poker) with ZenHub @apoorv-2204 @blackode @Neylix

**Comment by AmodD** [comment]
Please [add your planning poker estimate](https://app.zenhub.com/workspaces/Product-and-Design-Workspace-61efe71b9083a90018250d89/issues/archethic-foundation/archethic-node/303?planning-poker) with ZenHub @samuel-uniris

**Comment by apoorv-2204** [comment]
Given a situation, 
I am trying to mint nft color , which are just 256 color from (erc721 implemented smart contract)
So One utxo ID and one NFT ID (0to255) mapping ??
instead of One utxo id including all 256.?

**Comment by samuel-uniris** [comment]
> Given a situation, I am trying to mint nft color , which are just 256 color from (erc721 implemented smart contract) So One utxo ID and one NFT ID (0to255) mapping ?? instead of One utxo id including all 256.?

Yes

**Comment by apoorv-2204** [comment]
https://misc.archethic.net/fLXItn9yTU2P0-pb-rT9lw

**Comment by apoorv-2204** [comment]
> In Bitcoin UTXO model, to identify which UTXO from which transaction we are using, a transaction ID and UTXO ID is used.

tx_id -> txn hash, pointer to old txn whose output was this utxo
vout -> The index number of the UTXO to be spent; first one is 0

> The proposal would be to add an UTXO's ID and NFT's ID in the transaction data structure to resolve this point.

In ae -> the utxo_id will equivalent to vout(like in BTC)? 
in Array of unspent output `uco: %UCOLedger{transfers: [%UCOLedger.transfer{},| ,...] }`
 and `nft: %NFTedger{transfers: [%NFTLedger.transfer{}, | ...]}` -
  The array index will be depicting utxo_id.??

In ae: the nft_id , is added to just identify a nft from a particular set of nft's.

```
 %NFTLedger.Transfer{
        nft_id: binary()
       #  max limit? or no limit -> Number()
          nft: binary(),
          to: binary(),
          amount: non_neg_integer(),
          conditions: list(binary())
        }
```

**Comment by apoorv-2204** [comment]
It is Resolved by #390
